geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaFeatureWriter.scala

```diff
@@ -122,4 +124,13 @@ object KafkaFeatureWriter {
       producer.send(record)
     }
   }
+
+  trait RequiredVisibilityWriter extends AppendKafkaFeatureWriter {
```

Comment:
Just to make sure I'm following, these traits extend the class that they are mixed back into in order to override a method from the base class, right?

Would an alternative by to have this trait require a `feature: SimpleFeature =>` and then we could mix it in more easily?  (I think I'm thinking "cake pattern" ish...)

Reply:
i thought this was the cake pattern haha. but i wanted to keep it fairly tightly bound so that we're not executing "excess" code on each write.

Reply:
I guess I'm wondering if we can have one `RequiredVisibilityWriter` which extends something like `FeatureWriter` and then mixes in.

Such a class may need the little syntax for requiring a "feature: SimpleFeature" to be part of it, right?

Reply:
ah, maybe, but the one in index-api is hooked into IndexWriter not FeatureWriter. also, since feature is a var i'm not sure if that would work? i think it would have to be redefined with a function accessor.

Reply:
Eh, I think a var would work.  Do you want to try or leave it as-is?

Reply:
i could add it at the geomesa feature writer level, but the check here would double in size: https://github.com/locationtech/geomesa/blob/main/geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaFeatureWriter.scala#L101-L111
otherwise, i'd have to standardize some var names here and make these public/ maybe protected: https://github.com/locationtech/geomesa/blob/main/geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaFeatureWriter.scala#L225

i guess i'm not crazy about extracting it to the feature writer level b/c it relies on some low-level implementation details